### PR TITLE
Added yrd

### PR DESCRIPTION
**Pull request by makeworld-the-better-one**
This also adds a newer IPFS version, which is already done in develop. This was forked from master originally, if you want I can refork from develop.

### CODE REVIEW

File: scripts/install2 [hunk 1]
```diff
@@ -404,7 +415,10 @@ fi
 if [ ! -x "$(command -v ipfs)" ] && [ ! -z "$WITH_IPFS" -a "$WITH_IPFS" == "true" ]; then
     source ipfs/install
 fi
-
+# yrd
+if [ ! -x "$(command -v ipfs)" ] && [ ! -z "$WITH_YRD" -a "$WITH_YRD" == "true" ]; then
```
Comment (benhylau) [inline scripts/install2]: Remove the ipfs check.

File: scripts/install2 [hunk 2]
```diff
@@ -440,6 +454,7 @@ if [[ "$SUPPORT_WATCHDOG" == "true" ]] && [ ! -z "$WITH_WATCHDOG" -a "$WITH_WATC
     source watchdog/install
 fi
 
+
```
Comment (benhylau) [inline scripts/install2]: Why new newline?

File: scripts/yrd/install [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# get previous location - should be 
+START_DIR="$(pwd)"
+
+BASE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+mkdir "$BASEDIR/tmp"
+cd "$BASEDIR/tmp"
+git clone https://github.com/kpcyrd/yrd.git
+cd yrd
+# install dependencies
```
Comment (benhylau) [inline scripts/yrd/install]: For consistency with other files in this repo, comments are formatted like this:

`# Install dependencies`

With `#` then space then capital unless it's a name that starts with a lowercase letter (such as yrd).

File: scripts/yrd/uninstall [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```
Comment (benhylau) [inline scripts/yrd/uninstall]: See how ipfs/uninstall does it. Should disable, then stop, then daemon-reload, then remove. Also notice the `|| true` and `rm -f` flag. These prevent errors when things aren't running to being with when you uninstall.

```
# Uninstall IPFS
sudo systemctl disable ipfs.service 2>/dev/null || true
sudo systemctl stop ipfs.service 2>/dev/null || true
sudo systemctl daemon-reload
sudo rm -f /usr/local/bin/ipfs
sudo rm -f /etc/systemd/system/ipfs.service
```

File: scripts/yrd/install [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+sudo apt-get install python3 python3-pip python3-argh python3-requests
+# install yrd
+sudo -H python3 ./setup.py -v install
+# install and enable the service
+cd "$START_DIR"
+sudo cp yrd.service /etc/systemd/system/
+sudo systemctl enable yrd
```
Comment (benhylau) [inline scripts/yrd/install]: Let's add this before enabling to be safe:

```
sudo systemctl daemon-reload
```

Maybe it's no longer necessary but I have had issues with earlier images.

File: scripts/yrd/install [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -e
+
+# get previous location - should be 
+START_DIR="$(pwd)"
```
Comment (benhylau) [inline scripts/yrd/install]: I think this line should go after the `BASE_DIR` line.

Comment (darkdrgn2k) [reply scripts/yrd/install]: Don't think `START_DIR` is even needed. The place that it is used `BASE_DIR` should have been used for consistency

File: scripts/yrd/install [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+cd "$BASEDIR/tmp"
+git clone https://github.com/kpcyrd/yrd.git
+cd yrd
+# install dependencies
+sudo apt-get install python3 python3-pip python3-argh python3-requests
+# install yrd
+sudo -H python3 ./setup.py -v install
+# install and enable the service
+cd "$START_DIR"
+sudo cp yrd.service /etc/systemd/system/
```
Comment (darkdrgn2k) [inline scripts/yrd/install]: Don't need to change work directory.   And relaying on work directory is not the correct action as work directory can be different then where the file is located.

Instead use `BASE_DIR`
`sudo cp "$BASE_DIR/yrd.service" "/etc/systemd/system/yrd.service"`

**Comment by makeworld-the-better-one** [comment]
@benhylau @darkdrgn2k I made changes and pushed them to this branch. Can we merge now, or is there another step?

File: scripts/yrd/install [hunk 8]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -e
+
+BASE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+mkdir "$BASEDIR/tmp"
+cd "$BASEDIR/tmp"
+git clone https://github.com/kpcyrd/yrd.git
+cd yrd
+# Install dependencies
```
Comment (benhylau) [inline scripts/yrd/install]: Usually we put a blank line above each comment

File: scripts/yrd/uninstall [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+set -e
+
+# Uninstall yrd
+sudo systemctl disable yrd 2>/dev/null || true
+sudo systemctl stop yrd 2>/dev/null || true
+sudo systemctl daemon-reload
+sudo pip3 uninstall yrd
+sudo apt-get remove python3-argh python3-pip python3-requests python3
```
Comment (benhylau) [inline scripts/yrd/uninstall]: This isn't used by other stuff right?

**Pull request merged**